node/find.go

```diff
@@ -19,19 +19,66 @@ func (n *Node) GetSingleCidHandler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
+
 	log.Infow("Find cid", "cid", mhCid)
-	// Get Cid from primary storage
-	i, _, _ := n.primary.Get(c)
-	out := map[cid.Cid][]store.IndexEntry{c: i}
+	var (
+		entries []store.IndexEntry
+		found   bool
+	)
+
+	// Lookup CID in primary storage if primary enabled
+	if n.primary != nil {
+		entries, found, err = n.primary.Get(c)
+		if err != nil {
+			fmt.Println(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+	}
 
+	// Lookup CID in persistent storage if not found yet and persistent enabled
+	if !found && n.persistent != nil {
+		entries, found, err = n.primary.Get(c)
+		if err != nil {
+			fmt.Println(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		// Store the data that was found in persistent storage in primary storage
+		if found && n.primary != nil {
+			// TODO: Implement this:
```

Comment:
This is handled in my next PR. In the end I decoupled this from the API so we can use the combined storage (cache+persistence) also as a library. Thanks for the note.